codex-rs/tui/src/bottom_pane/mod.rs

```diff
@@ -80,6 +80,30 @@ impl BottomPane<'_> {
         }
     }
 
+    /// Handle Ctrl-C in the bottom pane. If a modal view is active it gets a
+    /// chance to consume the event (e.g. to dismiss itself). Returns true when
+    /// handled.
+    pub(crate) fn on_ctrl_c(&mut self) -> bool {
+        if let Some(mut view) = self.active_view.take() {
+            if view.on_ctrl_c(self) {
+                if !view.is_complete() {
+                    self.active_view = Some(view);
+                } else if self.is_task_running {
+                    self.active_view = Some(Box::new(StatusIndicatorView::new(
+                        self.app_event_tx.clone(),
+                    )));
+                }
+                // Show the standard Ctrl-C quit hint so a subsequent Ctrl-C will exit.
+                self.show_ctrl_c_quit_hint();
+                return true;
+            } else {
+                // Put the view back unchanged
+                self.active_view = Some(view);
+            }
+        }
+        false
+    }
```

Comment:
Taking the above suggestion into account, I would rearrange things slightly to leverage an early return and thereby reduce indenting a bit:

```suggestion
        let mut view = match self.active_view.take() {
            Some(view) => view,
            None => return CancellationEvent::Ignored,
        };
        
        let event = view.on_ctrl_c(self);
        match event {
            CancellationEvent::Handled => {
                if !view.is_complete() {
                    self.active_view = Some(view);
                } else if self.is_task_running {
                    self.active_view = Some(Box::new(StatusIndicatorView::new(
                        self.app_event_tx.clone(),
                    )));
                }
                // Show the standard Ctrl-C quit hint so a subsequent Ctrl-C will exit.
                self.show_ctrl_c_quit_hint();
            }
            CancellationEvent::Ignored => {
                // Put the view back unchanged
                self.active_view = Some(view);
            }
        }
        event
    }
```